library_generation/repo-level-postprocess/delete_non_generated_samples.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+set -e
+
+while IFS= read -r samples_dir; do
```

Comment:
Can we now stop the root cause of generating unnecessary samples?

Reply:
I think it related to pom templates in [synthtool](https://github.com/googleapis/synthtool/tree/master/synthtool/gcp/templates/java_library/samples).

@diegomarquezp Do you depend on pom.xml in this directory? Can I remove them?

Reply:
@JoeWang1127 No, I think the poms related to our work are in the [owlbot implementation](https://github.com/googleapis/sdk-platform-java/tree/main/library_generation/owlbot/templates/poms)

Reply:
@JoeWang1127 Did you get the answer how to stop generating unnecessary samples into monorepo?

Reply:
I think this issue can be solved by hacking into synthtool, for example change [_generic_library](https://github.com/googleapis/synthtool/blob/af16e6d4672cc7b400f144de2fc3068b54ff47d2/synthtool/gcp/common.py#L52).

However, should we do it now or after migrating all java specific functions to sdk-platform-java? @diegomarquezp what do you think of this problem?

Reply:
@JoeWang1127 We have a pending cleanup consisting of moving all the templates in `synthtool/synthtool` to `sdk-platform-java/library_generation/owlbot`. We initially meant to "move" the templates but we can also get rid of a few that are not used, such as `CONTRIBUTING.md` or the samples mentioned here. 

I would say this unnecessary-sample-fix should be done in that cleanup PR

Reply:
@suztomo Do you think it's reasonable to defer the refactor to further PRs?

Reply:
That's sounds good. Avoid copying `delete_non_generated_samples.sh`.

Reply:
We'll use the hermetic build script to generate google-cloud-java, so the current approach is to copy the script and refactor in later PR.

Reply:
Since I changed the template of `owlbot.py`, we don't need this script anymore.